client/src/components/CategoryExpenses.tsx

```diff
@@ -11,123 +12,193 @@ interface IProps {
 	eSetFilter: (term: string, match: string) => void
 }
 
-export class CategoryExpenses extends React.Component<IProps, {}> {
-	constructor(props: IProps) {
-		super(props)
+// @ts-ignore
+const CategoryExpenses: React.FC<IProps> = ({
+	categories,
+	sCategoryName,
+	eSetFilter,
+}) => {
+	const chartOptions = {
+		responsive: true,
+		maintainAspectRatio: false,
+		tooltipTemplate: '<%= label %>',
 	}
 
-	public render() {
-		const { categories, sCategoryName } = this.props
-
-		const chartOptions = {
-			responsive: true,
-			maintainAspectRatio: false,
-			tooltipTemplate: '<%= label %>',
-		}
+	// https://flatuicolors.com/palette/cn
+	// https://flatuicolors.com/palette/nl
+	const oCategoryColours: CategoryColors = {
+		accomodation: '#EA2027', // red - red pigment
+		food: '#2ed573', // green - ufo green
+		working: '#a4b0be', // grey - peace
+		recreation: '#1e90ff', // blue - clear chill
+		transport: '#5758BB', // lavendar - circumorbital ring
+		'non-food shopping': '#F79F1F', // orangish/gold - radiant yellow
+		miscellaneous: '#833471', // light purple - hollyhock
+		health: '#ff6b81', // pink - wild watermelon
+		utility: '#747d8c', // darker grey - bay wharf
+		giving: '#7bed9f', // light green - lime soap
+	}
 
-		// https://flatuicolors.com/palette/cn
-		// https://flatuicolors.com/palette/nl
-		const oCategoryColours: CategoryColors = {
-			accomodation: '#EA2027', // red - red pigment
-			food: '#2ed573', // green - ufo green
-			working: '#a4b0be', // grey - peace
-			recreation: '#1e90ff', // blue - clear chill
-			transport: '#5758BB', // lavendar - circumorbital ring
-			'non-food shopping': '#F79F1F', // orangish/gold - radiant yellow
-			miscellaneous: '#833471', // light purple - hollyhock
-			health: '#ff6b81', // pink - wild watermelon
-			utility: '#747d8c', // darker grey - bay wharf
-			giving: '#7bed9f', // light green - lime soap
+	const chartData = categories.map(oP => {
+		return {
+			value: Number(oP.percent).toFixed(0),
+			label: `${oP.category}: $${Number(oP.total).toFixed(0)} - ${Number(
+				oP.percent,
+			).toFixed(0)}%`,
+			color: oCategoryColours[oP.category]
+				? oCategoryColours[oP.category]
+				: `hsla(${Math.random() * 360}, 100%, 50%, 1)`,
 		}
+	})
 
-		const chartData = categories.map(oP => {
-			return {
-				value: Number(oP.percent).toFixed(0),
-				label: `${oP.category}: $${Number(oP.total).toFixed(
-					0,
-				)} - ${Number(oP.percent).toFixed(0)}%`,
-				color: oCategoryColours[oP.category]
-					? oCategoryColours[oP.category]
-					: `hsla(${Math.random() * 360}, 100%, 50%, 1)`,
-			}
-		})
+	// @ts-ignore
```

Comment:
don't ts-ignore this, I just tried typing it as discussed and it works. type it with Column from the lib.